src/main.cpp

```diff
@@ -33,63 +37,136 @@ constexpr auto VERSION = _VERSION;
 #else
 constexpr auto VERSION = "[Unkown version]";
 #endif
+class InputParser {
+ public:
+  InputParser(int& argc, char** argv) {
+    for (int i = 1; i < argc; ++i) this->tokens.push_back(std::string(argv[i]));
```

Comment:
You can use `split` from `utils/string.h` with ' ' as delimiter.

Reply:
I'm not exactly sure what you're suggesting, but perhaps you mean replacing this for loop with a call to split? If so then I based on my memory of that func it doesn't make sense as we're not actually parsing a single string here, just storing the args in a cpp vector (going from memory here too).

I'll take a look at this & get back to you.

Reply:
OK, after looking at your string.h split does the job!

Reply:
After going to change this I realized my last comment was wrong.😱 Split does a similar job, but in this case the args are already split & we're just putting them into a vector from the c array.